devito/symbolics/manipulation.py

```diff
@@ -523,3 +527,14 @@ def unevaluate(expr):
         return uneval_mapper[expr.func](*args)
     except KeyError:
         return reuse_if_untouched(expr, args)
+
+
+def as_long(expr):
+    """
+    Convert an expression and its symbolic args to a long integer.
```

Comment:
Actually, I just checked, as long as one of the operands is cast to `long`, you're good to go. IOW, you don't need to cast all operands to long. Which might improve verbosity. 

So technically we might get away with simply just changing these lines

```
SizeOf(obj._C_typedata)*obj.size
```

into

```
ULONG(SizeOf(obj._C_typedata))*obj.size
```

Sorry, I only thought about this now 😬

Reply:
No it doesn't, I actually tried with the JIT backdoor and the `obj.size` will evaluate as int first then converted to long so if it overflows then the conversion will not do anything